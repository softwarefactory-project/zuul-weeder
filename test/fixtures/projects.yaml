- project:
    name: config
    check:
      jobs:
        - config-check
    gate:
      jobs:
        - config-check
    post:
      jobs:
        - config-update

- project:
    templates:
      - sf-ci-jobs
    check:
      jobs:
        - linters:
            nodeset: linters-pod
    gate:
      jobs:
        - linters:
            nodeset: linters-pod
    experimental:
      jobs:
        - sf-rpm-build
        - sf-ci-openshift-integration:
            dependencies:
              - sf-rpm-build