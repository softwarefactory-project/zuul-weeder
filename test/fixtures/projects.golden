[ ZProjectPipeline
    ( ProjectPipeline
        { name = PName
            ( ProjectName "config" )
        , templates = []
        , pipelines = fromList
            [ PPipeline
                { name = PipelineName "check"
                , jobs =
                    [ PJName
                        ( JobName "config-check" )
                    ]
                }
            , PPipeline
                { name = PipelineName "gate"
                , jobs =
                    [ PJName
                        ( JobName "config-check" )
                    ]
                }
            , PPipeline
                { name = PipelineName "post"
                , jobs =
                    [ PJName
                        ( JobName "config-update" )
                    ]
                }
            ]
        }
    )
, ZProjectPipeline
    ( ProjectPipeline
        { name = PNameCannonical
            ( CanonicalProjectName
                ( ProviderName ""
                , ProjectName ""
                )
            )
        , templates =
            [ ProjectTemplateName "sf-ci-jobs" ]
        , pipelines = fromList
            [ PPipeline
                { name = PipelineName "check"
                , jobs =
                    [ PJJob
                        ( Job
                            { name = JobName "linters"
                            , parent = Nothing
                            , nodeset = Just
                                ( JobNodeset
                                    ( NodesetName "linters-pod" )
                                )
                            , branches = []
                            , dependencies = []
                            }
                        )
                    ]
                }
            , PPipeline
                { name = PipelineName "experimental"
                , jobs =
                    [ PJName
                        ( JobName "sf-rpm-build" )
                    , PJJob
                        ( Job
                            { name = JobName "sf-ci-openshift-integration"
                            , parent = Nothing
                            , nodeset = Nothing
                            , branches = []
                            , dependencies =
                                [ JobName "sf-rpm-build" ]
                            }
                        )
                    ]
                }
            , PPipeline
                { name = PipelineName "gate"
                , jobs =
                    [ PJJob
                        ( Job
                            { name = JobName "linters"
                            , parent = Nothing
                            , nodeset = Just
                                ( JobNodeset
                                    ( NodesetName "linters-pod" )
                                )
                            , branches = []
                            , dependencies = []
                            }
                        )
                    ]
                }
            ]
        }
    )
]